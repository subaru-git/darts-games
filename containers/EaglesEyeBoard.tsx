import React, { FC } from "react";
import { Grid, Table, TableContainer, Tbody, Td, Tr } from "@chakra-ui/react";

type EaglesEyeBoardProps = {
  data: number[][];
};

const EaglesEyeBoard: FC<EaglesEyeBoardProps> = ({ data }) => {
  return (
    <div>
      <Grid>
        <TableContainer>
          <Table variant="simple">
            <Tbody>
              {data.map((round, i) => (
                <Tr key={`round-${round}-${i}`}>
                  {round.map((score, j) => (
                    <Td key={`score-${score}-${j}`}>
                      {score === 50 ? "D-BULL" : score === 25 ? "S-BULL" : "-"}
                    </Td>
                  ))}
                </Tr>
              ))}
            </Tbody>
          </Table>
        </TableContainer>
      </Grid>
    </div>
  );
};

export default EaglesEyeBoard;
