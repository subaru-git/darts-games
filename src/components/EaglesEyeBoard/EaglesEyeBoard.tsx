import React, { FC } from 'react'
import { Grid, Table, TableContainer, Tbody, Td, Tr } from '@chakra-ui/react'

type EaglesEyeBoardProps = {
  data: point[][]
}

const EaglesEyeBoard: FC<EaglesEyeBoardProps> = ({ data }) => {
  return (
    <Grid>
      <TableContainer>
        <Table variant='simple'>
          <Tbody>
            {data.map((round, i) => (
              <Tr key={`round-${round}-${i}`}>
                {round.map((score, j) => (
                  <Td key={`score-${score}-${j}`}>{score.includes('BULL') ? score : '-'}</Td>
                ))}
              </Tr>
            ))}
          </Tbody>
        </Table>
      </TableContainer>
    </Grid>
  )
}

export default EaglesEyeBoard
