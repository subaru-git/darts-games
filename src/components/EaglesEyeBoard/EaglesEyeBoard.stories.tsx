/* eslint-disable import/named */
import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import EaglesEyeBoard from './EaglesEyeBoard';

export default {
  title: 'Components/EaglesEyeBoard',
  component: EaglesEyeBoard,
} as ComponentMeta<typeof EaglesEyeBoard>;

const Template: ComponentStory<typeof EaglesEyeBoard> = (args) => <EaglesEyeBoard {...args} />;

export const Default = Template.bind({});
Default.args = {
  data: [
    ['20T', 'D-BULL', 'S-BULL'],
    ['D-BULL', '19D', '1T'],
  ],
};
