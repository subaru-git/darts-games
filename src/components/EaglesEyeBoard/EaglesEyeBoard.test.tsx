import { render, screen } from '@testing-library/react';
import EaglesEyeBoard from './EaglesEyeBoard';

test('should rendering', () => {
  const { container } = render(<EaglesEyeBoard data={[['D-BULL', '0', 'S-BULL']]} />);
  expect(screen.getAllByText('D-BULL')).toHaveLength(1);
  expect(screen.getAllByText('-')).toHaveLength(1);
  expect(screen.getAllByText('S-BULL')).toHaveLength(1);
  expect(container).toMatchSnapshot();
});
