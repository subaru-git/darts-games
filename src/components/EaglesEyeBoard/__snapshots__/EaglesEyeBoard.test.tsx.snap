// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="css-lgj0h8"
  >
    <div
      class="chakra-table__container css-zipzvv"
    >
      <table
        class="chakra-table css-0"
      >
        <tbody
          class="css-0"
        >
          <tr
            class="css-0"
          >
            <td
              class="css-0"
            >
              D-BULL
            </td>
            <td
              class="css-0"
            >
              -
            </td>
            <td
              class="css-0"
            >
              S-BULL
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
